argonOne/run.sh

```diff
@@ -54,40 +54,40 @@ fanSpeedReport(){
       *)
         icon=mdi:fan;
     esac
-    reqBody='{"state": "'"${percent}"'", "attributes": { "unit_of_measurement": "%", "icon": "'"${icon}"'", "mode": "'"${mode}"'", "Temperature '"${CorF}"'": "'"${temp}"'", "fan level": "'"${level}"'", "friendly_name": "Argon Fan Speed"}}'
-    nc -i 1 hassio 80 1>/dev/null <<<unix2dos<<EOF
+    reqBody='{"state": "'"${fanPercent}"'", "attributes": { "unit_of_measurement": "%", "icon": "'"${icon}"'", "mode": "'"${mode}"'", "Temperature '"${CorF}"'": "'"${temp}"'", "fan level": "'"${level}"'", "friendly_name": "Argon Fan Speed"}}'
+    nc -i 1 hassio 80 1>/dev/null <<< unix2dos<<EOF
 POST /homeassistant/api/states/sensor.argon_one_addon_fan_speed HTTP/1.1
 Authorization: Bearer ${SUPERVISOR_TOKEN}
 Content-Length: $( echo -ne "${reqBody}" | wc -c )
 
 ${reqBody}
 EOF
-};
+}
 
 
 
 action() {
-  percent=${1}
+  fanPercent=${1}
   level=${2}
   name=${3}
-  percentHex=${4}
-  temp=${5}
-  CorF=${6}
+  temp=${4}
+  CorF=${5}
+  fanPercentHex=$(printf '%x' "${fanPercent}")
```

Comment:
There was no reason to pass the `fanPercentHex` as an argument, as it is used only here. Therefore you can rather just do it here instead